src/components/API.js

```diff
@@ -0,0 +1,120 @@
+export default class Api {
+  constructor({ baseURL, headers }) {
+    this._baseURL = baseURL;
+    this._headers = headers;
+  }
+
+  /* ------------------ Checks Response from Server ------------------ */
+  _checkServerResponse(res) {
+    return res.ok ? res.json() : Promise.reject(`Error: ${res.status}`);
+  }
+
+  /* ------------------ Retrieves UserInfo from Server ------------------ */
+  getUserInfo() {
+    return fetch(`${this._baseURL}/users/me`, {
+      headers: this._headers,
+    })
+      .then((res) => {
+        return this._checkServerResponse(res);
+      })
+      .catch((err) => {
+        console.log(err);
+      });
```

Comment:
It’s better not to use `catch` in `Api` because you need to `catch` possible errors at the very end of any request and this end is in `index.js`   after all blocks `then`. And in real projects blocks `catch` usually open error popups and so on.

Please, remove it from `Api` and add it in `index.js`.